### PR TITLE
Add API response tests from TIP examples

### LINKED ISSUE #1663: Add test parsing all API response examples from the TIP
https://github.com/iotaledger/tips/pull/153

### PR DESCRIPTION
**Pull request by thibault-martinez**
Fixes https://github.com/iotaledger/iota-sdk/issues/1663

### CODE REVIEW

File: sdk/tests/types/api/core.rs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    // GET /api/core/v3/blocks/{blockId}/full
+    json_response::<BlockWithMetadataResponse>("get-full-block-by-id-tagged-data-response-example.json").unwrap();
+    // GET /api/core/v3/outputs/{outputId}
+    json_response::<OutputResponse>("get-outputs-by-id-response-example.json").unwrap();
+    // GET /api/core/v3/outputs/{outputId}/metadata
+    json_response::<OutputMetadata>("get-output-metadata-by-id-response-unspent-example.json").unwrap();
+    json_response::<OutputMetadata>("get-output-metadata-by-id-response-spent-example.json").unwrap();
+    // GET /api/core/v3/outputs/{outputId}/full
+    json_response::<OutputWithMetadata>("get-full-output-metadata-example.json").unwrap();
+    // GET /api/core/v3/transactions/{transactionId}/metadata
```
Comment (Thoralf-M) [inline sdk/tests/types/api/core.rs:94]: What about `/api/core/v3/transactions/{transactionId}/included-block` and `/api/core/v3/transactions/{transactionId}/included-block/metadata`?

Comment (Thoralf-M) [reply sdk/tests/types/api/core.rs:94]: I guess it's covered by `// GET /api/core/v3/blocks/{blockId}/metadata`, just not sure where you put the API routes in the comments and where not

Comment (thibault-martinez) [reply sdk/tests/types/api/core.rs:94]: They reuse the same example as block and block/metadata

**Review by Thoralf-M — APPROVED** [review]

**Review by DaughterOfMars — COMMENTED** [review]
Care about the missing newlines in the json files?

**Comment by thibault-martinez** [comment]
> Care about the missing newlines in the json files?

no

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**